gradle/jmh.gradle

```diff
@@ -0,0 +1,38 @@
+//
+// Configure JMH benchmarks
+//
+
+sourceSets {
+    jmh {
+        compileClasspath += sourceSets.main.output
+        runtimeClasspath += sourceSets.main.output
+    }
+}
+
+task jmh(type: JavaExec, description: 'Executing JMH benchmarks', group: 'performance') {
+    classpath = sourceSets.jmh.runtimeClasspath
+    main = 'org.openjdk.jmh.Main'
+
+    def profilers = ['pauses']
```

Comment:
I'll take a look at this, it does seem like there could be something done. What values, other than `pauses` would be acceptable?

Reply:
The list for other profilers is specific to what the host can provide I think. Can't run with a `perf` profiler unless you have `perf` installed.
There's another gradle task `jmhProfilers` which lists them. Pauses is a nice safe extra but it would be nice if people could add some extras from time to time. Maybe not something to commit for everyone else though?

Reply:
I've had a look and the nicest thing I came up with was converting those two tasks to be in well-defined classes. It's a lot more wordy now, but I'd argue it's "nicer".

However, I also found https://github.com/melix/jmh-gradle-plugin exists - maybe it would be worth using that instead of the custom tasks?

Reply:
Given the use of JMH right now is minimal at best, probably not worth spending too much time and effort here until the need comes up.
A separate issue can be raised if people want changes to the jmh task etc.